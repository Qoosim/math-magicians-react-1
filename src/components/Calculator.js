import React from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [data, setData] = React.useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (value) => {
    setData((prevData) => calculate(prevData, value));
  };

  return (
    <main>
      <section className="cal-section">
        <h3>
          Are you ready to challenge your skill?
          <br />
          Let us do some mathematics.
        </h3>
        <div className="calc-grid">
          <div className="output-div">
            <div className="output" data-testid="output">
              {
                (data.total ? data.total : '')
                  + (data.operation ? data.operation : '')
                  + (data.next ? data.next : '') || 0
              }
            </div>
          </div>
          <button type="button" onClick={() => handleClick('AC')}>AC</button>
          <button type="button" onClick={() => handleClick('+/-')}>+/-</button>
          <button type="button" onClick={() => handleClick('%')}>%</button>
          <button type="button" onClick={() => handleClick('รท')} className="operation">รท</button>
          <button type="button" onClick={() => handleClick('7')}>7</button>
          <button type="button" onClick={() => handleClick('8')}>8</button>
          <button type="button" onClick={() => handleClick('9')}>9</button>
          <button type="button" onClick={() => handleClick('x')} className="operation">x</button>
          <button type="button" onClick={() => handleClick('4')}>4</button>
          <button type="button" onClick={() => handleClick('5')}>5</button>
          <button type="button" onClick={() => handleClick('6')}>6</button>
          <button type="button" onClick={() => handleClick('-')} className="operation">-</button>
          <button type="button" onClick={() => handleClick('1')}>1</button>
          <button type="button" onClick={() => handleClick('2')}>2</button>
          <button type="button" onClick={() => handleClick('3')}>3</button>
          <button type="button" onClick={() => handleClick('+')} className="operation">+</button>
          <button type="button" onClick={() => handleClick('0')} className="span-two">0</button>
          <button type="button" onClick={() => handleClick('.')}>.</button>
          <button type="button" onClick={() => handleClick('=')} className="operation">=</button>
        </div>
      </section>
    </main>
  );
};

export default Calculator;
