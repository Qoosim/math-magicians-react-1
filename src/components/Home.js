import React from 'react';
import Nav from './Nav';

const Home = () => (
  <main className="home-container">
    <Nav />
    <section>
      <h3>Welcome to the home of mathematics</h3>
      <p>
        Lorem ipsum dolor sit amet consectetur, adipisicing elit.
        Asperiores quaerat cumque magni perspiciatis quam distinctio,
        natus eaque saepe iure! Molestias facere, libero recusandae
        sapiente rerum consectetur officiis at hic saepe ipsa voluptate
        ipsum tenetur voluptatum quia eaque quaerat ullam laborum nisi consequatur,
        aut labore, quod neque. Illo dolorum sapiente vero reprehenderit inventore,
        iste in culpa asperiores ad tenetur molestias illum velit maxime voluptatem
        eum repudiandae eligendi, nisi eaque explicabo, suscipit pariatur.
        Sed hic pariatur explicabo inventore? Rerum tenetur sapiente, dolorum sequi
        harum dolores officia! Aliquid, magni in a consequuntur, atque animi provident,
        labore est placeat velit nemo illo asperiores expedita!
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur, adipisicing elit.
        Asperiores quaerat cumque magni perspiciatis quam distinctio,
        natus eaque saepe iure! Molestias facere, libero recusandae
        sapiente rerum consectetur officiis at hic saepe ipsa voluptate
        ipsum tenetur voluptatum quia eaque quaerat ullam laborum nisi consequatur,
        aut labore, quod neque. Illo dolorum sapiente vero reprehenderit inventore,
        iste in culpa asperiores ad tenetur molestias illum velit maxime voluptatem
        eum repudiandae eligendi, nisi eaque explicabo, suscipit pariatur.
        Sed hic pariatur explicabo inventore? Rerum tenetur sapiente, dolorum sequi
        harum dolores officia! Aliquid, magni in a consequuntur, atque animi provident,
        labore est placeat velit nemo illo asperiores expedita!
      </p>
    </section>
  </main>
);

export default Home;
