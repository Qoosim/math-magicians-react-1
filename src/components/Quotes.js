import React from 'react';

const Quotes = () => (
  <main>
    <section>
      <h1 className="quote">Qoute of the day</h1>
      <p className="quote-msg">
        Mathematics is not about numbers, equations, computations or algorithm:
        It is about understanding.
        <br />
        **-William Paul Thurston-**
      </p>
    </section>
  </main>
);

export default Quotes;
