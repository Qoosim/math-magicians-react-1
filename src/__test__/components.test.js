import React from 'react';
import '@testing-library/jest-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import Home from '../components/Home';
import Calculator from '../components/Calculator';
import Quotes from '../components/Quotes';

describe('Tests for components', () => {
  test('renders Home snapshot', () => {
    const tree = render(<Home />);
    expect(tree).toMatchSnapshot();
  });

  test('Calculator snapshot', () => {
    const tree = render(<Calculator />);
    expect(tree).toMatchSnapshot();
  });

  test('Quotes snapshot', () => {
    const tree = render(<Quotes />);
    expect(tree).toMatchSnapshot();
  });

  test('Calculator component test', () => {
    const tree = render(<Calculator />);
    expect(tree).toMatchSnapshot();
    fireEvent.click(screen.getByText('2'));
    expect(screen.getByTestId('output')).toHaveTextContent('2');
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByTestId('output')).toHaveTextContent('9');
  });
});
