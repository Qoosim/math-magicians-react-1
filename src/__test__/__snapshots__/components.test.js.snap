// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for components Calculator component test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <section
          class="cal-section"
        >
          <h3>
            Are you ready to challenge your skill?
            <br />
            Let us do some mathematics.
          </h3>
          <div
            class="calc-grid"
          >
            <div
              class="output-div"
            >
              <div
                class="output"
                data-testid="output"
              >
                0
              </div>
            </div>
            <button
              type="button"
            >
              AC
            </button>
            <button
              type="button"
            >
              +/-
            </button>
            <button
              type="button"
            >
              %
            </button>
            <button
              class="operation"
              type="button"
            >
              รท
            </button>
            <button
              type="button"
            >
              7
            </button>
            <button
              type="button"
            >
              8
            </button>
            <button
              type="button"
            >
              9
            </button>
            <button
              class="operation"
              type="button"
            >
              x
            </button>
            <button
              type="button"
            >
              4
            </button>
            <button
              type="button"
            >
              5
            </button>
            <button
              type="button"
            >
              6
            </button>
            <button
              class="operation"
              type="button"
            >
              -
            </button>
            <button
              type="button"
            >
              1
            </button>
            <button
              type="button"
            >
              2
            </button>
            <button
              type="button"
            >
              3
            </button>
            <button
              class="operation"
              type="button"
            >
              +
            </button>
            <button
              class="span-two"
              type="button"
            >
              0
            </button>
            <button
              type="button"
            >
              .
            </button>
            <button
              class="operation"
              type="button"
            >
              =
            </button>
          </div>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <section
        class="cal-section"
      >
        <h3>
          Are you ready to challenge your skill?
          <br />
          Let us do some mathematics.
        </h3>
        <div
          class="calc-grid"
        >
          <div
            class="output-div"
          >
            <div
              class="output"
              data-testid="output"
            >
              0
            </div>
          </div>
          <button
            type="button"
          >
            AC
          </button>
          <button
            type="button"
          >
            +/-
          </button>
          <button
            type="button"
          >
            %
          </button>
          <button
            class="operation"
            type="button"
          >
            รท
          </button>
          <button
            type="button"
          >
            7
          </button>
          <button
            type="button"
          >
            8
          </button>
          <button
            type="button"
          >
            9
          </button>
          <button
            class="operation"
            type="button"
          >
            x
          </button>
          <button
            type="button"
          >
            4
          </button>
          <button
            type="button"
          >
            5
          </button>
          <button
            type="button"
          >
            6
          </button>
          <button
            class="operation"
            type="button"
          >
            -
          </button>
          <button
            type="button"
          >
            1
          </button>
          <button
            type="button"
          >
            2
          </button>
          <button
            type="button"
          >
            3
          </button>
          <button
            class="operation"
            type="button"
          >
            +
          </button>
          <button
            class="span-two"
            type="button"
          >
            0
          </button>
          <button
            type="button"
          >
            .
          </button>
          <button
            class="operation"
            type="button"
          >
            =
          </button>
        </div>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tests for components Calculator snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <section
          class="cal-section"
        >
          <h3>
            Are you ready to challenge your skill?
            <br />
            Let us do some mathematics.
          </h3>
          <div
            class="calc-grid"
          >
            <div
              class="output-div"
            >
              <div
                class="output"
                data-testid="output"
              >
                0
              </div>
            </div>
            <button
              type="button"
            >
              AC
            </button>
            <button
              type="button"
            >
              +/-
            </button>
            <button
              type="button"
            >
              %
            </button>
            <button
              class="operation"
              type="button"
            >
              รท
            </button>
            <button
              type="button"
            >
              7
            </button>
            <button
              type="button"
            >
              8
            </button>
            <button
              type="button"
            >
              9
            </button>
            <button
              class="operation"
              type="button"
            >
              x
            </button>
            <button
              type="button"
            >
              4
            </button>
            <button
              type="button"
            >
              5
            </button>
            <button
              type="button"
            >
              6
            </button>
            <button
              class="operation"
              type="button"
            >
              -
            </button>
            <button
              type="button"
            >
              1
            </button>
            <button
              type="button"
            >
              2
            </button>
            <button
              type="button"
            >
              3
            </button>
            <button
              class="operation"
              type="button"
            >
              +
            </button>
            <button
              class="span-two"
              type="button"
            >
              0
            </button>
            <button
              type="button"
            >
              .
            </button>
            <button
              class="operation"
              type="button"
            >
              =
            </button>
          </div>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <section
        class="cal-section"
      >
        <h3>
          Are you ready to challenge your skill?
          <br />
          Let us do some mathematics.
        </h3>
        <div
          class="calc-grid"
        >
          <div
            class="output-div"
          >
            <div
              class="output"
              data-testid="output"
            >
              0
            </div>
          </div>
          <button
            type="button"
          >
            AC
          </button>
          <button
            type="button"
          >
            +/-
          </button>
          <button
            type="button"
          >
            %
          </button>
          <button
            class="operation"
            type="button"
          >
            รท
          </button>
          <button
            type="button"
          >
            7
          </button>
          <button
            type="button"
          >
            8
          </button>
          <button
            type="button"
          >
            9
          </button>
          <button
            class="operation"
            type="button"
          >
            x
          </button>
          <button
            type="button"
          >
            4
          </button>
          <button
            type="button"
          >
            5
          </button>
          <button
            type="button"
          >
            6
          </button>
          <button
            class="operation"
            type="button"
          >
            -
          </button>
          <button
            type="button"
          >
            1
          </button>
          <button
            type="button"
          >
            2
          </button>
          <button
            type="button"
          >
            3
          </button>
          <button
            class="operation"
            type="button"
          >
            +
          </button>
          <button
            class="span-two"
            type="button"
          >
            0
          </button>
          <button
            type="button"
          >
            .
          </button>
          <button
            class="operation"
            type="button"
          >
            =
          </button>
        </div>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tests for components Quotes snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <section>
          <h1
            class="quote"
          >
            Qoute of the day
          </h1>
          <p
            class="quote-msg"
          >
            Mathematics is not about numbers, equations, computations or algorithm: It is about understanding.
            <br />
            **-William Paul Thurston-**
          </p>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <section>
        <h1
          class="quote"
        >
          Qoute of the day
        </h1>
        <p
          class="quote-msg"
        >
          Mathematics is not about numbers, equations, computations or algorithm: It is about understanding.
          <br />
          **-William Paul Thurston-**
        </p>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tests for components renders Home snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="home-container"
      >
        <section>
          <h3>
            Welcome to the home of mathematics!
          </h3>
          <p>
            Lorem ipsum dolor sit amet consectetur, adipisicing elit. Asperiores quaerat cumque magni perspiciatis quam distinctio, natus eaque saepe iure! Molestias facere, libero recusandae sapiente rerum consectetur officiis at hic saepe ipsa voluptate ipsum tenetur voluptatum quia eaque quaerat ullam laborum nisi consequatur, aut labore, quod neque. Illo dolorum sapiente vero reprehenderit inventore, iste in culpa asperiores ad tenetur molestias illum velit maxime voluptatem eum repudiandae eligendi, nisi eaque explicabo, suscipit pariatur. Sed hic pariatur explicabo inventore? Rerum tenetur sapiente, dolorum sequi harum dolores officia! Aliquid, magni in a consequuntur, atque animi provident, labore est placeat velit nemo illo asperiores expedita!
          </p>
          <p>
            Lorem ipsum dolor sit amet consectetur, adipisicing elit. Asperiores quaerat cumque magni perspiciatis quam distinctio, natus eaque saepe iure! Molestias facere, libero recusandae sapiente rerum consectetur officiis at hic saepe ipsa voluptate ipsum tenetur voluptatum quia eaque quaerat ullam laborum nisi consequatur, aut labore, quod neque. Illo dolorum sapiente vero reprehenderit inventore, iste in culpa asperiores ad tenetur molestias illum velit maxime voluptatem eum repudiandae eligendi, nisi eaque explicabo, suscipit pariatur. Sed hic pariatur explicabo inventore? Rerum tenetur sapiente, dolorum sequi harum dolores officia! Aliquid, magni in a consequuntur, atque animi provident, labore est placeat velit nemo illo asperiores expedita!
          </p>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="home-container"
    >
      <section>
        <h3>
          Welcome to the home of mathematics!
        </h3>
        <p>
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Asperiores quaerat cumque magni perspiciatis quam distinctio, natus eaque saepe iure! Molestias facere, libero recusandae sapiente rerum consectetur officiis at hic saepe ipsa voluptate ipsum tenetur voluptatum quia eaque quaerat ullam laborum nisi consequatur, aut labore, quod neque. Illo dolorum sapiente vero reprehenderit inventore, iste in culpa asperiores ad tenetur molestias illum velit maxime voluptatem eum repudiandae eligendi, nisi eaque explicabo, suscipit pariatur. Sed hic pariatur explicabo inventore? Rerum tenetur sapiente, dolorum sequi harum dolores officia! Aliquid, magni in a consequuntur, atque animi provident, labore est placeat velit nemo illo asperiores expedita!
        </p>
        <p>
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Asperiores quaerat cumque magni perspiciatis quam distinctio, natus eaque saepe iure! Molestias facere, libero recusandae sapiente rerum consectetur officiis at hic saepe ipsa voluptate ipsum tenetur voluptatum quia eaque quaerat ullam laborum nisi consequatur, aut labore, quod neque. Illo dolorum sapiente vero reprehenderit inventore, iste in culpa asperiores ad tenetur molestias illum velit maxime voluptatem eum repudiandae eligendi, nisi eaque explicabo, suscipit pariatur. Sed hic pariatur explicabo inventore? Rerum tenetur sapiente, dolorum sequi harum dolores officia! Aliquid, magni in a consequuntur, atque animi provident, labore est placeat velit nemo illo asperiores expedita!
        </p>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
